src/python/pants/backend/go/build_integration_test.py

```diff
@@ -35,116 +36,85 @@ def rule_runner() -> RuleRunner:
-    source_files_str = ", ".join(f'"{sf.path}"' for sf in source_files)
-    deps_str = ", ".join(f'"{addr.spec}"' for addr in dependencies)
-    rule_runner.add_to_build_file(
-        "",
-        f"go_package(name='{target_name}', import_path='{import_path}', sources=[{source_files_str}], dependencies=[{deps_str}])\n",
-    )
-    return rule_runner.get_target(Address("", target_name=target_name))
-
-
-def build_package(
-    rule_runner: RuleRunner,
-    main_target: Target,
-) -> BuiltPackage:
-    args = ["--backend-packages=pants.backend.go"]
-    rule_runner.set_options(args)
-    rule_runner.add_to_build_file(
-        "", f"go_binary(name='bin', binary_name='foo', main='{main_target.address.spec}')\n"
-    )
-    go_binary_target = rule_runner.get_target(Address("", target_name="bin"))
-    built_package = rule_runner.request(BuiltPackage, (GoBinaryFieldSet.create(go_binary_target),))
-    return built_package
-
-
-def test_package_one_target(rule_runner: RuleRunner) -> None:
+def test_package_simple(rule_runner: RuleRunner) -> None:
     rule_runner.write_files(
-        {"BUILD": "go_module(name='root')\n", "go.mod": "module foo.example.com\n"}
+        {
+            "go.mod": "module foo.example.com\n",
+            "BUILD": "go_module(name='root')\n",
+            "project/main.go": dedent(
+                """\
+                package main
+
+                import (
+                    "fmt"
+                )
+
+                func main() {
+                    fmt.Println("Hello world!")
+                }
+                """
+            ),
+            "project/BUILD": dedent(
+                """\
+                go_package(name='main', import_path='main')
+                go_binary(name='bin', binary_name='foo', main=':main')
+                """
+            ),
+        }
     )
-    target = make_target(rule_runner, [MAIN_SOURCE], import_path="main")
-    built_package = build_package(rule_runner, target)
+    binary_tgt = rule_runner.get_target(Address("project", target_name="bin"))
+    built_package = build_package(rule_runner, binary_tgt)
     assert len(built_package.artifacts) == 1
-    assert built_package.artifacts[0].relpath == "foo"
+    assert built_package.artifacts[0].relpath == "project/foo"
 
 
-def test_package_with_depenedency(rule_runner: RuleRunner) -> None:
+def test_package_with_dependency(rule_runner: RuleRunner) -> None:
     rule_runner.write_files(
-        {"BUILD": "go_module(name='root')\n", "go.mod": "module foo.example.com\n"}
-    )
-    lib_target = make_target(
-        rule_runner, [LIB_SOURCE], import_path="example.com/lib", target_name="lib"
-    )
-    main_target = make_target(
-        rule_runner,
-        [MAIN_SOURCE],
-        import_path="main",
-        dependencies=(lib_target.address,),
-        target_name="main",
+        {
+            "go.mod": "module foo.example.com\n",
+            "BUILD": "go_module(name='root')\n",
+            "project/lib/lib.go": dedent(
```

Comment:
I'm still working through Go's official tutorial so have a very nascent understanding of module vs package...Is this right @tdyas?

Reply:
style nit: Go code for a small module would be put directly in the module root (so `go_module` and `go_package` in same directory). If the code had to be put in a subdirectory, many Go packages would choose `pkg` as the name.

Reply:
or in this case, just have `lib/iib.go` without the `project` part.

Reply:
The reason I have a subdir is because we want to test `dependencies`. We expect there to be one `go_package` per directory, so we need a subdir. 

Is this new setup right? i.e. `lib/lib.go` which sets the import path `foo.example.com/lib`?

Reply:
> The reason I have a subdir is because we want to test dependencies. We expect there to be one go_package per directory, so we need a subdir.

Having a subdir is fine, but having two subdirs (`project` and `lib`) seemed unnecessary given how Go code is usually structured.

> Is this new setup right? i.e. lib/lib.go which sets the import path foo.example.com/lib?

Looks good. And correct, `lib/lib.go` has an import path of `foo.example.com/lib` given the module has an import path of `foo.example.com`.